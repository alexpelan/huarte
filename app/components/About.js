import React from 'react';
import {
  StatusBar,
  View,
  Text,
} from 'react-native';

import Link from './Link';
import styles from '../styles/styles';

class About extends React.Component {
  render() {
    StatusBar.setBarStyle('default', true);
    return (
      <View style={[styles.loadingView, styles.paragraphView]}>
        <Text style={styles.loadingText}>
          Huarte is a game by <Link text="Alex Pelan," url="http://www.alexpelan.com" /> a software developer in Brooklyn, NY. The questions and answers are from <Link text="J-Archive." url="www.j-archive.com" /> The app was built using Facebook&apos;s <Link text="React Native" url="https://facebook.github.io/react-native/" /> framework and its source is fully available <Link text="here." url="https://github.com/alexpelan/huarte" />
        </Text>
      </View>
    );
  }
}

export default About;
