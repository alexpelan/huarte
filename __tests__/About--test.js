import 'react-native';
import React from 'react';
import About from '../app/components/About';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <About />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});