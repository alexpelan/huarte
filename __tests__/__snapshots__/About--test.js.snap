// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#0000af",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "paddingLeft": 20,
        "paddingRight": 20,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 24,
      }
    }
  >
    Huarte is a game by 
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "textDecorationLine": "underline",
        }
      }
    >
      Alex Pelan,
    </Text>
     a software developer in Brooklyn, NY. The questions and answers are from 
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "textDecorationLine": "underline",
        }
      }
    >
      J-Archive.
    </Text>
     The app was built using Facebook's 
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "textDecorationLine": "underline",
        }
      }
    >
      React Native
    </Text>
     framework and its source is fully available 
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "textDecorationLine": "underline",
        }
      }
    >
      here.
    </Text>
  </Text>
</View>
`;
